git/odb/file_storer.go

```diff
@@ -50,8 +50,22 @@ func (fs *fileStorer) Store(sha []byte, r io.Reader) (n int64, err error) {
 	if err != nil {
 		return n, err
 	}
+	if err = tmp.Close(); err != nil {
+		return n, err
+	}
 
 	path := fs.path(sha)
+	dir := filepath.Dir(path)
+
+	if _, err := os.Stat(dir); os.IsNotExist(err) {
+		// Since .git/objects partitions objects based on the first two
+		// characters of their ASCII-encoded SHA1 object ID, ensure that
+		// the directory exists before copying a file into it.
+		err = os.MkdirAll(dir, 0755)
```

Comment:
Is there a benefit to calling `os.Stat()` before `os.MkdirAll()`? According to the [docs](https://golang.org/pkg/os/#MkdirAll):

> If path is already a directory, MkdirAll does nothing and returns nil.